### PR TITLE
AUTH-482: set required-scc for openshift workloads

### PR DESCRIPTION
**Pull request by liouk**

### CODE REVIEW

**Comment by liouk** [comment]
Interestingly, the `oauth-openshift` pods were getting admitted with the `node-exporter` SCC (belongs to the [cluster-monitoring-operator](https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/node-exporter/security-context-constraints.yaml)) exactly because of the problem that AUTH-482 describes.

**Comment by liouk** [comment]
/hold
need to verify SCC

File: manifests/07_deployment.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
         app: authentication-operator
       annotations:
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+        openshift.io/required-scc: anyiud
```
Comment (ibihim) [inline manifests/07_deployment.yaml:25]: `anyuid` is sufficient? Interesting. I've seen that in the cluster as well.

**Comment by ibihim** [comment]
Could be a lgtm, but the tests look failed... 🤔

**Review by stlaz — CHANGES_REQUESTED** [review]

File: pkg/operator/assets/bindata.go [hunk 2]
```diff
@@ -230,6 +230,7 @@ spec:
         apiserver: "true"
       annotations:
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+        openshift.io/required-scc: privileged
```
[the remark this answers is not in the document]
Comment (liouk) [reply pkg/operator/assets/bindata.go]: Done in 76183e3db5ef956316fff6280e0c2c69e1f0a5b9

File: bindata/assets.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package bindata
+
+import (
+	"embed"
+)
+
+//go:embed oauth-apiserver/* oauth-apiserver/RBAC/* oauth-openshift/*
```
Comment (stlaz) [inline bindata/assets.go:7]: //go:embed oauth-apiserver/* oauth-openshift/*

**Comment by stlaz** [comment]
https://github.com/openshift/cluster-authentication-operator/blob/6214f1b9fc57d14d242eb4af8c31da727a61582a/pkg/dependencymagnet/dependencymagnet.go#L9
this dep magnet also seems to be ready to be removed

**Comment by liouk** [comment]
debug commits dropped
/hold cancel

**Pull request merged**